File: apnaghar-frontend/src/services/propertyService.js
import api from "./api";

// Get paginated properties
export const getAllProperties = (page = 0, size = 5) => {
  return api.get(`/properties?page=${page}&size=${size}`);
};

// Get single property by ID
export const getPropertyById = (id) => {
  return api.get(`/properties/${id}`);
};

// Search by location
export const searchByLocation = (location) => {
  return api.get(`/properties/search/location?location=${location}`);
};

// Search by type
export const searchByType = (type) => {
  return api.get(`/properties/search/type?type=${type}`);
};

// Search by location + type
export const searchByLocationAndType = (location, type) => {
  return api.get(`/properties/search?location=${location}&type=${type}`);
};

// Add new property (OWNER)
export const addProperty = (property, ownerEmail) => {
  return api.post(`/properties?ownerEmail=${ownerEmail}`, property);
};

File: src/main/java/com/apnaghar/service/PropertyService.java
package com.apnaghar.service;

import com.apnaghar.model.Property;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface PropertyService {

    // ✅ Owner adds property
    Property addProperty(Property property, String ownerEmail);

    // ✅ Existing (keep it)
    List<Property> getAllProperties();

    // 🔥 NEW – pagination support
    Page<Property> getAllProperties(Pageable pageable);

    // ✅ Search by location
    List<Property> searchByLocation(String location);

    // ✅ Search by type
    List<Property> searchByType(String type);

    // ✅ Search by location + type
    List<Property> search(String location, String type);
    
    Property getPropertyById(Long id);

    List<Property> getMyProperties(String ownerEmail);

    //this is for delete property
    void deleteProperty(Long propertyId, String ownerEmail);

    // this is for update feature of property
    Property updateProperty(Long propertyId, Property updatedProperty, String ownerEmail);

}
